File: src/main/java/ru/hollowhorizon/mastertech/api/IProxy.java
package ru.hollowhorizon.mastertech.api;

import net.minecraft.item.Item;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

public interface IProxy {
    void preInit(FMLPreInitializationEvent e);

    void init(FMLInitializationEvent e);

    void postInit(FMLPostInitializationEvent e);

    void getRender(Item item, int meta, String id);

    default void getDefaultItemRenderer(Item item) {}
}

File: src/main/java/ru/hollowhorizon/mastertech/api/IProxy.kt
package ru.hollowhorizon.mastertech.api

import net.minecraft.item.Item
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

interface IProxy {
    fun preInit(e: FMLPreInitializationEvent)

    fun init(e: FMLInitializationEvent)

    fun postInit(e: FMLPostInitializationEvent)

    fun getRender(item: Item, meta: Int, id: String)

    fun getDefaultItemRenderer(item: Item) {}
}
